internal/blockchain/difficulty.go

```diff
@@ -57,11 +57,17 @@ func (b *BlockChain) calcNextRequiredDifficulty(prevNode *blockNode, newBlockTim
 
 	// We're not at a retarget point, return the oldDiff.
 	params := b.chainParams
-	if (prevNode.height+1)%params.WorkDiffWindowSize != 0 {
-		// For networks that support it, allow special reduction of the
-		// required difficulty once too much time has elapsed without
-		// mining a block.
-		if params.ReduceMinDifficulty {
+	nextHeight := prevNode.height + 1
+	if nextHeight%params.WorkDiffWindowSize != 0 {
+		// For networks that support it, allow special reduction of the required
+		// difficulty once too much time has elapsed without mining a block.
+		//
+		// Note that this behavior is deprecated and thus is only supported on
+		// testnet v3 prior to the max diff activation height.  It will be
+		// removed in future version of testnet.
+		if params.ReduceMinDifficulty && (!b.isTestNet3() || nextHeight <
```

Comment:
Feels like a comment is warranted in chaincfg/testnetparams.go where `ReduceMinDifficulty` is actually set to call out this caveat.

Reply:
Good point.  I did deprecate it, but it's worth calling out on the parameter too in the mean time.  Updated.